### ISSUE TITLE
Search for products fails when attached to sync plan.

### ISSUE THREAD

**Issue by cliles** [SatelliteQE/nailgun#339]
In using the katello ansible module, I have found some failures when searching for a product when said product is assigned to a sync plan.

```
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_xUYjnd/ansible_module_katello.py", line 533, in <module>
    main()
  File "/tmp/ansible_xUYjnd/ansible_module_katello.py", line 510, in main
    result = ng.repository_set(params)
  File "/tmp/ansible_xUYjnd/ansible_module_katello.py", line 293, in repository_set
    product = self.find_product(params['product'], params['organization'])
  File "/tmp/ansible_xUYjnd/ansible_module_katello.py", line 176, in find_product
    response = product.search()
  File "/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/blinker_herald/base.py", line 137, in wrapper
    result = fn(*args, **kwargs)
  File "/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/nailgun/entity_mixins.py", line 1299, in search
    results = self.search_normalize(results)
  File "/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/nailgun/entity_mixins.py", line 1202, in search_normalize
    attrs[field_name] = _get_entity_id(field_name, result)
  File "/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/nailgun/entity_mixins.py", line 247, in _get_entity_id
    return attrs[field_name]['id']
KeyError: 'id'
```

The module searches like so
```
def find_product(self, name, organization):
        org = self.find_organization(organization)

        product = self._entities.Product(self._server, name=name, organization=org)
        response = product.search()
```

If I follow the documentation for nailgun, this should search and return all products that match name and org. It looks like entity_mixins._get_entity_id() is not safe for this case.

**Comment by oshtaier** [comment]
Yes, you are right.
Just tried to make similar search:
```
>>> product = entities.Product(name='ZcMWYxqhkX', organization=22).search()
>>> product
[nailgun.entities.Product(nailgun.config.ServerConfig(url=u'server', verify=False, auth=(u'admin', u'changeme')), gpg_key=None, description=None, label=u'm7deYidx5W', organization=nailgun.entities.Organization(nailgun.config.ServerConfig(url=u'server', verify=False, auth=(u'admin', u'changeme')), id=22), sync_plan=None, id=178, name=u'ZcMWYxqhkX')]
```
but once sync plan is assigned, I got:
```
product = entities.Product(name='ZcMWYxqhkX', organization=22).search()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/site-packages/blinker_herald/base.py", line 137, in wrapper
    result = fn(*args, **kwargs)
  File "nailgun/entity_mixins.py", line 1299, in search
    results = self.search_normalize(results)
  File "nailgun/entity_mixins.py", line 1202, in search_normalize
    attrs[field_name] = _get_entity_id(field_name, result)
  File "nailgun/entity_mixins.py", line 247, in _get_entity_id
    return attrs[field_name]['id']
KeyError: 'id'
```
Let's me try to play a little with it

**Comment by abalakh** [comment]
Most likely #261 and #262 are related. Haven't analyzed yet, but it may appear #262 fixed `read()` for product with sync plan and we need something similar for `search()` too.

**Comment by oshtaier** [comment]
Yes, I got to the same point in debug for search functionality. It doesn't seem right to fix on our side, but we can add a workaround for it

**Comment by jbnance** [comment]
+1 for a workaround for this

**Comment by ehelms** [comment]
I am hitting this in a new way with latest code:

```
Traceback (most recent call last):
  File "./blah", line 20, in <module>
    response = product.search()
  File "/usr/lib/python2.7/site-packages/blinker_herald/base.py", line 137, in wrapper
    result = fn(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/nailgun/entity_mixins.py", line 1362, in search
    for result in results
  File "/usr/lib/python2.7/site-packages/nailgun/entities.py", line 4344, in __init__
    super(Product, self).__init__(server_config, **kwargs)
  File "/usr/lib/python2.7/site-packages/nailgun/entity_mixins.py", line 414, in __init__
    self._server_config
  File "/usr/lib/python2.7/site-packages/nailgun/entity_mixins.py", line 137, in _make_entity_from_id
    return entity_cls(server_config, id=entity_obj_or_id)
  File "/usr/lib/python2.7/site-packages/nailgun/entities.py", line 5861, in __init__
    _check_for_value('organization', kwargs)
  File "/usr/lib/python2.7/site-packages/nailgun/entities.py", line 156, in _check_for_value
    'A value must be provided for the "{0}" field.'.format(field_name)
TypeError: A value must be provided for the "organization" field.
```

**Comment by itewk** [comment]
Did anyone ever find a solution to this? I am running into this now.

**Comment by cliles** [comment]
@itewk - This issue is so old, I have since moved on. In the end I had to fall back to scripting around hammer.

**Comment by cliles** [comment]
I am back again hitting this with the same error as @ehelms!

**Comment by abalakh** [comment]
Fixed in #582
